src/tasks.coffee

```diff
@@ -0,0 +1,314 @@
+TaskModel = require('./model/tasks').Task
+Channel = require('./model/channels').Channel
+Q = require("q")
+logger = require("winston")
+config = require("./config/config")
+config.rerun = config.get('rerun')
+http = require 'http'
+TransactionModel = require("./model/transactions").Transaction
+net = require "net"
+rerunMiddleware = require "./middleware/rerunUpdateTransactionTask"
+
+
+live = false
+activeTasks = 0
+
+
+findAndProcessAQueuedTask = ->
+  TaskModel.findOneAndUpdate { status: 'Queued' }, { status: 'Processing' }, (err, task) ->
+    if err
+      logger.err "An error occurred while looking for rerun tasks: #{err}"
+    else if task
+      activeTasks++
+      processNextTaskRound task, (err) ->
+        logger.err "An error occurred while processing rerun task #{task._id}: #{err}" if err
+        activeTasks--
+        # task has finished its current round, pick up the next one
+        if live then findAndProcessAQueuedTask()
+
+      # task has started processing, pick up the next one
+      if live then findAndProcessAQueuedTask()
+
+rerunTaskProcessor = ->
+  if live
+    findAndProcessAQueuedTask()
+    setTimeout rerunTaskProcessor, config.rerun.processor.pollPeriodMillis
+
+
+exports.start = (callback) ->
+  live = true
+  setTimeout rerunTaskProcessor, config.rerun.processor.pollPeriodMillis
+
+  logger.info "Started rerun task processor"
+  callback()
+
+
+exports.stop = (callback) ->
+  live = false
+
+  waitForActiveTasks = ->
```

Comment:
Not sure I follow - So the setTimeout triggers the polling process. We could use promises to check for active tasks, rather the counter; is that what you mean? It might just be a bit tricky to do it that way right now, and this works.